base_transaction_id/models/invoice.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2019-2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    transaction_id = fields.Char(
+        string="Transaction ID",
+        index=True,
+        copy=False,
+        store=True,
+        help="Transaction ID from the financial institute",
+    )
+
+    def _get_computed_reference(self):
+        if self.transaction_id:
+            return self.transaction_id
+        return super()._get_computed_reference()
```

Comment:
I've just found that `_get_computed_reference` had been changed to `_get_invoice_computed_reference` with new helpers. So the code should be updated accordingly. (https://github.com/odoo/odoo/commit/bc131c0cfb51c953de8ec41fb820c8c7831eefb5)
```suggestion
    def _get_invoice_reference_odoo_invoice(self):
        if self.transaction_id:
            return self.transaction_id
        return super()._get_invoice_reference_odoo_invoice()
```